### DISCUSSION TITLE
Use key instead of id?

### DISCUSSION THREAD

**Discussion by GollyJer** [urql-graphql/urql#1534] [Q&A] [answered]
Our data uses `key` for all id's. 
Is there an easy way to tell urql to use `key` instead of `id` for the cacheExchange?

**Comment by kitten** [comment, marked as answer]
I assume you mean you're using Graphcache and are looking for this? https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/#custom-keys-and-non-keyable-entities

**Reply by GollyJer** [reply]
Thanks for the reply @kitten.
Yes, I'm talking about Graphcache.

Basically what I'm asking is if there's an easy way to tell Graphcache to add `key` to it's list of default id's.
So instead of `id, _id` it will use `id, _id, key`?

If I'm understanding correctly you're saying we need to continue to do this for all our keys?
```javascript 
cacheExchange({
  keys: {
    Thing1: data => data.key,
    Thing2: data => data.key,
    Thing3: data => data.key
    etc...
  },
});
`
